sdk/src/main/scala/com/horizen/certificatesubmitter/CertificateSubmitter.scala

```diff
@@ -159,8 +160,10 @@ class CertificateSubmitter
 
     val actualPoseidonRootHash =
       BackwardTransferLoader.schnorrFunctions.generatePoseidonHash(backwardTransferPublicKeys.map(_.bytes()).asJava, params.backwardTransferThreshold)
-    if (actualPoseidonRootHash.deep != params.poseidonRootHash.deep) {
-      throw new IllegalStateException(s"Incorrect configuration for backward transfer, expected poseidon root hash ${params.poseidonRootHash.deep} but actual is ${actualPoseidonRootHash.deep}")
+
+    val expectedPoseidonRootHash = getSidechainCreationTransaction(sidechainNodeView.history).getBackwardTransferPoseidonRootHash
```

Comment:
to be discussed. It can be like "params.genesisBlock.mainchainReferences.head.sidechainRelatedAggregatedTransaction."
We can trust the value stored in params.